include/zephyr/bluetooth/audio/hap.h

```diff
@@ -0,0 +1,394 @@
+	 *
+	 * Optional callback called when the active preset is changed by the remote server when the
+	 * preset switch procedure is complete. The callback must be set to receive active preset
+	 * changes and enable support for switching presets.
+	 *
+	 * @param harc Controller instance pointer.
+	 * @param index Active preset index.
+	 */
+	void (*active)(struct bt_hap_harc *harc, uint8_t index);
+
+	/**
+	 * @brief Callback function for preset store.
+	 *
+	 * The callback is called when the Preset Record Update ot Preset Read Response is notified
+	 * by the remote server.
+	 * The record object as well as its objects are temporary and must be copied to in order
+	 * to cache its information.
+	 *
+	 * @param harc Controller instance pointer.
+	 * @param record Preset record to store.
+	 */
+	void (*store)(struct bt_hap_harc *harc, const struct bt_has_preset_record *record);
+
+	/**
+	 * @brief Callback function for preset deletion.
+	 *
+	 * The callback provides a range of presets that have been deleted by the remote server.
+	 *
+	 * @param conn Bluetooth connection object.
+	 * @param start_index Preset start index that has been deleted.
+	 * @param end_index Preset end index that has been deleted.
+	 */
+	void (*remove)(struct bt_hap_harc *harc, uint8_t start_index, uint8_t end_index);
+
+	/**
+	 * @brief Callback function for preset availability notifications.
+	 *
+	 * The callback is called when the preset availability change is notified by the remote
+	 * server.
+	 *
+	 * @param harc Controller instance pointer.
+	 * @param index Preset index.
+	 * @param available True if available, false otherwise.
+	 */
+	void (*available)(struct bt_hap_harc *harc, uint8_t index, bool available);
+
+	/**
+	 * @brief Preset list update complete.
+	 *
+	 * Function called after all the preset list changes have been submitted.
+	 * User might apply the updated list to the application.
+	 *
+	 * @param harc Controller instance pointer.
+	 */
+	void (*commit)(struct bt_hap_harc *harc);
+
+	/**
+	 * @brief Preset get callback function.
+	 *
+	 * The callback is called to retrieve the stored preset record.
+	 *
+	 * @param harc Controller instance pointer.
+	 * @param index Preset index.
+	 * @param[out] record Preset record object to fill in.
+	 */
+	int (*get)(struct bt_hap_harc *harc, uint8_t index, struct bt_has_preset_record *record);
+};
+
+/**
+ * @brief Registers Hearing Aid Remote Controller Preset List callbacks.
+ *
+ * @param cb The callback structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_hap_harc_preset_cb_register(const struct bt_hap_harc_preset_cb *cb);
+
+/**
+ * @brief Get information for the profile connection.
+ *
+ * @param harc      Controller instance pointer.
+ * @param[out] info Info object.
+ *
+ * @return Zero on success or (negative) error code on failure.
+ */
+int bt_hap_harc_info_get(struct bt_hap_harc *harc, struct bt_hap_harc_info *info);
```

Comment:
```suggestion
int bt_hap_harc_info_get(const struct bt_hap_harc *harc, struct bt_hap_harc_info *info);
```

This should be possible to be `const`